Authentication/SignUp/View/SignupController.swift

```diff
@@ -0,0 +1,230 @@
+        if let passwordValidationMessageLabel = signupView.passwordValidationMessageLabel {
+            passwordValidationMessageLabel.rex_text <~ _viewModel.passwordValidationErrors.signal.map { $0.first ?? " " }
+        }
+        
+        _viewModel.passwordConfirmation <~ signupView.passwordConfirmTextField.rex_textSignal
+        signupView.passwordConfirmLabel?.text = _viewModel.confirmPasswordText
+        signupView.passwordConfirmTextField.placeholder = _viewModel.confirmPasswordPlaceholderText
+        _viewModel.passwordConfirmationValidationErrors.signal.observeNext { [unowned self] errors in
+            self._delegate.signupController(self, didFailPasswordConfirmationValidationWithErrors: errors)
+        }
+        if let passwordConfirmValidationMessageLabel = signupView.passwordConfirmValidationMessageLabel {
+            passwordConfirmValidationMessageLabel.rex_text <~ _viewModel.passwordConfirmationValidationErrors.signal.map { $0.first ?? " " }
+        }
+    }
+    
+    func bindButtons() {
+        signupView.signupButton.setTitle(_viewModel.signupButtonTitle, forState: .Normal)
+        signupView.signupButton.rex_pressed.value = _viewModel.signUpCocoaAction
+        signupView.signupButton.rex_enabled.signal.observeNext { [unowned self] in self.signupView.signupButtonEnabled = $0 }
+        
+        signupView.termsAndServicesLabel?.text = _viewModel.termsAndServicesLabelText
+        signupView.termsAndServicesButton.setTitle(_viewModel.termsAndServicesButtonTitle, forState: .Normal)
+        //signupView.termsAndServicesButton -> Presents the terms and services
+    }
+    
+}
+
+extension SignupController: UITextFieldDelegate {
+    
+    public func textFieldShouldReturn(textField: UITextField) -> Bool {
+        if textField == signupView.usernameTextField {
+            signupView.emailTextField.becomeFirstResponder()
+        } else if textField == signupView.emailTextField {
+            signupView.passwordTextField.becomeFirstResponder()
+        } else if textField == signupView.passwordTextField {
+            signupView.passwordConfirmTextField.becomeFirstResponder()
+        } else {
+            (signupView.usernameTextField ?? signupView.emailTextField).becomeFirstResponder()
+        }
+        return true
+    }
+    
+    public func textFieldDidBeginEditing(textField: UITextField) {
+        if textField == signupView.usernameTextField {
+            signupView.usernameTextFieldSelected = true
+        } else if textField == signupView.emailTextField {
+            signupView.emailTextFieldSelected = true
+        } else if textField == signupView.passwordTextField {
+            signupView.passwordTextFieldSelected = true
+        } else {
+            signupView.passwordConfirmationTextFieldSelected = true
+        }
+        _activeTextField.value = textField
+    }
+    
+    public func textFieldDidEndEditing(textField: UITextField) {
+        if textField == signupView.usernameTextField {
```

Comment:
I thought that we could have a boolean objc associated property "selected" to avoid having it in the view (just an idea though).
This would be reduced to 
```swift
public func textFieldDidEndEditing(textField: UITextField) {
    textField.selected = false
    _activeTextField.value = .None
}
```